src/lib/server/functions/getGraphData.ts

```diff
@@ -0,0 +1,86 @@
+import prisma from "$lib/server/database.js";
+import type { GraphMetrics } from "@prisma/client";
+import type { ChartConfiguration } from "chart.js";
+import dayjs from "dayjs";
+import { inPlaceSort } from "fast-sort";
+
+export default async function getGraphData(category: string, user?: string) {
+  let userItemCount: GraphMetrics[] = [];
+
+  if (user && user.match(/^\d{17,19}$/)) {
+    const privacyCheck = await prisma.preferences.findUnique({
+      where: { userId: user },
+      select: { leaderboards: true },
+    });
+
+    if (!privacyCheck) return "invalid user";
+
+    if (privacyCheck?.leaderboards) {
+      userItemCount = await prisma.graphMetrics.findMany({
+        where: {
+          AND: [
+            { userId: user },
+            { date: { gte: dayjs().subtract(45, "days").toDate() } },
+            { category: category },
+          ],
+        },
+      });
+    } else return "private profile"
+  } else return "invalid user";
+
+  if (userItemCount.length < 2) return "not enough data";
+
+  const userItemCounts = new Map<string, number>();
+
+  for (const item of userItemCount) {
+    userItemCounts.set(dayjs(item.date).format("YYYY-MM-DD"), Number(item.value));
+  }
+
+  const graphData: ChartConfiguration = {
+    type: "line",
+    data: {
+      labels: [],
+      datasets: [
+        {
+          yAxisID: "y2",
```

Comment:
Any reason why it’s not y1?

Reply:
Whenever it's y1 it shows money of the left, however  it doesnt show when hovering over and looks kind of strange with it all zeros on the right side. I don't want to have to redo the entire graph, but if you want to make it look better you can remove the left side numbers and have y1 and y2 just control whether dollar signs show up on the right

Reply:
<img width="1062" alt="Screenshot 2023-07-16 at 9 43 25 AM" src="https://github.com/tekoh/nypsi-website/assets/52770460/964aed4c-a23e-4921-af6e-f7c313161513">
example